src/models/Scholarships.ts

```diff
@@ -155,6 +159,16 @@ class Scholarships extends FirestoreCollection<ScholarshipData> {
   ): FirestoreModel<ScholarshipData> {
     return new FirestoreModel<ScholarshipData>(this.collection.doc(), data);
   }
+
+  /**
+   *
+   * @param reqs is the scholarship requirements e.g. (grades, states, etc.)
+   * @param paramFilters
+   * @returns whether a scholarship's requirements are included in the given filters
+   */
+  includesFilter(reqs?: any[], paramFilters?: any[]): boolean {
+    return !paramFilters || !reqs || reqs.some((r) => paramFilters.includes(r));
```

Comment:
You should handle the cases where either array is empty:
```suggestion
    return !paramFilters?.length || !reqs?.length || reqs.some((r) => paramFilters.includes(r));
```

Probably need some test cases for them.

Reply:
initially I didn't think it was necessary since we prune empty arrays, but I think it's still a good check to have.